products/views.py

```diff
@@ -1,3 +1,37 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+
+from .models import Product, Category
+
+class ProductListView(View):
+    def get(self, request, cat_id=None):
+        result = []
+
+        # 전체리스트
+        if cat_id == None:
+            products = Product.objects.all()
+
+        #카테고리별
+        else:   
+            products = Product.objects.filter(category_id=cat_id)
+
+            if not Category.objects.filter(id=cat_id).exists():
+                return JsonResponse({'message':'CATEGORY_DOES_NOT_EXIST'}, status=400)
```

Comment:
```suggestion
        products = Product.objects.all()
        
        if category_id:
            if not Category.objects.filter(id=cat_id).exists():
                return JsonResponse({'message':'CATEGORY_DOES_NOT_EXIST'}, status=400)
            products = products.filter(category_id=cat_id)
```
</br>
위와 같이 굳이 else문 없이도 작성하더라도 똑같이 동작합니다. 정민님이면 충분히 이해하실거라 생각합니다!
(설명이 필요하다면 QnA시간에 찾아와주세요 ㅎㅎ)

추가로 python은 interpreter 언어이기 때문에 코드를 한줄 한줄 읽으며 실행합니다.

`product = Product.objects.filter(category_id=cat_id)` 를 통해 DB hit를 하기전에 해당 쿼리셋이 존재하는지를 확인하는 바로 밑의 if문이 선행된다면 불필요한 DB조회가 줄어들 것입니다.